code/test-utils/src/main/java/com/adobe/marketing/mobile/util/NetworkRequestHelper.kt

```diff
@@ -26,7 +26,7 @@ import java.util.concurrent.TimeUnit
  */
 class NetworkRequestHelper {
 	private val sentTestableNetworkRequests: MutableMap<TestableNetworkRequest, MutableList<TestableNetworkRequest>>
-	private val networkResponses: MutableMap<TestableNetworkRequest, MutableList<HttpConnecting>>
+	private val networkResponses: MutableMap<TestableNetworkRequest, HttpConnecting>
```

Comment:
How do we handle the case where I set the expectation of 3 network requests to interact endpoint and I want to assert on all 3 responses?

Reply:
The use-case I see may use consecutive requests in a single test and you want to assert the order of execution and content for each, for example:
1. Expect 3 requests to interact
2. Send 3 consecutive personalization requests for 3 surfaces
3. Assert 3 requests happened in the same order as the API calls, and the corresponding response is valid for each

I see this more as an use-case for integration tests, for mocks keeping one mock response connection makes sense to me.

Reply:
We probably don't need to combine the requests and responses since we also keep track of recorded requests internally with sentTestableNetworkRequests. I think the simplest structure would be MutableMap<TestableNetworkRequest, MutableList<HttpConnecting>> so you take advantage of the equals implementation on the request. What do you think?